src/compiler/scala/tools/nsc/classpath/ZipArchiveFileLookup.scala

```diff
@@ -50,11 +50,15 @@ trait ZipArchiveFileLookup[FileEntryType <: ClassRepresentation] extends Efficie
     } yield createFileEntry(entry)
 
   protected def file(inPackage: PackageName, name: String): Option[FileEntryType] =
-    for {
-      dirEntry <- findDirEntry(inPackage)
-      entry <- Option(dirEntry.lookupName(name, directory = false))
-      if isRequiredFileType(entry)
-    } yield createFileEntry(entry)
+    findDirEntry(inPackage) match {
+      case Some(dirEntry) =>
+        val entry = dirEntry.lookupName(name, directory = false)
+        if (entry != null && isRequiredFileType(entry))
+          Some(createFileEntry(entry))
+        else
+          None
+      case _ => None
+    }
```

Comment:
Since `Option.flatMap` is inlined, I wonder if it is cheap enough to avoid pattern match idiom. (Similarly `map` where it would apply.) Since this codebase is optimized.